### PR TITLE
add constant for name, update EntryResolver to packit/planner

### PR DESCRIPTION
**Pull request by AccraZed**
## Summary
Addressed [this issue](https://github.com/paketo-buildpacks/dotnet-core-sdk/issues/172) regarding this buildpack using redundant entry resolver code that's been provided by [packit/planner](https://github.com/paketo-buildpacks/packit/draft)

## Use Cases
Cuts down on redundant code and reduces maintenance overhead

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by fg-j** [comment]
It would be useful to test in`build_test.go` that the draft.Planner's `MergeLayerTypes()` method is being used properly in the build.

If you remove [lines 63 and 64 of `build_test.go`](https://github.com/AccraZed/dotnet-core-sdk/blob/dc3bdbf9d7ae3fa444b5c237619be0442ab3d4fd/build_test.go#L63-L64), you'll see the unit tests fail in a way that reveals the lack of a faked call to `MergeLayerTypes()`.

Please update the test setup to directly fake the return of the MergeLayerTypes call.

File: build_test.go [hunk 1]
```diff
@@ -197,6 +198,7 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 					},
 				},
 			}))
+		Expect(entryResolver.MergeLayerTypesCall.CallCount).To(Equal(1))
 
 		Expect(dependencyManager.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbDir, "buildpack.toml")))
```
Comment (fg-j) [inline build_test.go:203]: To test that the MergeLayerTypes function is being used correctly in `build.go`, you can make assertions about the arguments passed to the function call. Like this line that tests an argument passed to `dependencyManager.Resolve()`.
```
		Expect(dependencyManager.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbDir, "buildpack.toml")))
```
If you test what the entryResolver MergeLayerTypes call _receives_, you don't need to separately test its call count in this case, because if call count = 0, the `Receives` field will have its zero value. That way, one assertion checks a) `MergeLayerTypes()` is called and b) it's called with appropriate arguments.

**Review by fg-j — APPROVED** [review]
